packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -965,3 +964,33 @@ class _ScrollingFlameChartRowExtentDelegate extends ExtentDelegate {
     return index;
   }
 }
+
+abstract class FlameChartPainter extends CustomPainter {
+  FlameChartPainter({
+    @required this.zoom,
+    @required this.constraints,
+    @required this.verticalScrollOffset,
+    @required this.horizontalScrollOffset,
+    @required this.chartStartInset,
+  }) : visible = Rect.fromLTWH(
+          horizontalScrollOffset,
+          verticalScrollOffset,
+          constraints.maxWidth,
+          constraints.maxHeight,
+        );
+
+  static const origin = 0.0;
```

Comment:
nit: consider just writing `0.0` and removing this constant unless you expect this constant will need to be changed in the future.